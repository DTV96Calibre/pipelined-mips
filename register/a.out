#! /usr/remote/bin/vvp
:ivl_version "10.1 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_0x12bba10 .scope module, "fetch_pipeline_reg" "fetch_pipeline_reg" 2 8;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "clear"
    .port_info 2 /INPUT 32 "pc_plus_four_F"
    .port_info 3 /INPUT 32 "instruction_F"
    .port_info 4 /OUTPUT 32 "pc_plus_four_D"
    .port_info 5 /OUTPUT 32 "instruction_D"
o0x7f77672781f8 .functor BUFZ 1, C4<z>; HiZ drive
v0x12e3750_0 .net "clear", 0 0, o0x7f77672781f8;  0 drivers
o0x7f7767278018 .functor BUFZ 1, C4<z>; HiZ drive
v0x12e3860_0 .net "clock", 0 0, o0x7f7767278018;  0 drivers
v0x12e39b0_0 .net "instruction_D", 31 0, v0x12e1e40_0;  1 drivers
o0x7f7767278228 .functor BUFZ 32, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0x12e3a50_0 .net "instruction_F", 31 0, o0x7f7767278228;  0 drivers
v0x12e3af0_0 .net "pc_plus_four_D", 31 0, v0x12e2de0_0;  1 drivers
o0x7f77672784c8 .functor BUFZ 32, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0x12e3c30_0 .net "pc_plus_four_F", 31 0, o0x7f77672784c8;  0 drivers
S_0x12ba1b0 .scope module, "instruction" "pipeline_reg" 2 18, 3 9 0, S_0x12bba10;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "clear"
    .port_info 2 /INPUT 32 "new_value"
    .port_info 3 /OUTPUT 32 "curr_value"
L_0x7f776722f0a8 .functor BUFT 1, C4<00000000000000000000000000000000>, C4<0>, C4<0>, C4<0>;
v0x12e2260_0 .net/2u *"_s0", 31 0, L_0x7f776722f0a8;  1 drivers
v0x12e2360_0 .net "clear", 0 0, o0x7f77672781f8;  alias, 0 drivers
v0x12e2420_0 .net "clock", 0 0, o0x7f7767278018;  alias, 0 drivers
v0x12e24f0_0 .net "curr_value", 31 0, v0x12e1e40_0;  alias, 1 drivers
v0x12e25c0_0 .net "gated_new_value", 31 0, L_0x12f3f70;  1 drivers
v0x12e26b0_0 .net "new_value", 31 0, o0x7f7767278228;  alias, 0 drivers
L_0x12f3f70 .functor MUXZ 32, o0x7f7767278228, L_0x7f776722f0a8, o0x7f77672781f8, C4<>;
S_0x12bc9d0 .scope module, "r" "register" 3 19, 4 11 0, S_0x12ba1b0;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "should_write"
    .port_info 2 /INPUT 32 "new_value"
    .port_info 3 /OUTPUT 32 "curr_value"
v0x12bb280_0 .net "clock", 0 0, o0x7f7767278018;  alias, 0 drivers
v0x12e1e40_0 .var "curr_value", 31 0;
v0x12e1f20_0 .var "is_init", 0 0;
v0x12e1ff0_0 .net "new_value", 31 0, L_0x12f3f70;  alias, 1 drivers
L_0x7f776722f0f0 .functor BUFT 1, C4<1>, C4<0>, C4<0>, C4<0>;
v0x12e20d0_0 .net "should_write", 0 0, L_0x7f776722f0f0;  1 drivers
E_0x12ca520 .event posedge, v0x12bb280_0;
S_0x12e2800 .scope module, "pc_plus_four" "pipeline_reg" 2 17, 3 9 0, S_0x12bba10;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "clear"
    .port_info 2 /INPUT 32 "new_value"
    .port_info 3 /OUTPUT 32 "curr_value"
L_0x7f776722f018 .functor BUFT 1, C4<00000000000000000000000000000000>, C4<0>, C4<0>, C4<0>;
v0x12e31d0_0 .net/2u *"_s0", 31 0, L_0x7f776722f018;  1 drivers
v0x12e32d0_0 .net "clear", 0 0, o0x7f77672781f8;  alias, 0 drivers
v0x12e3390_0 .net "clock", 0 0, o0x7f7767278018;  alias, 0 drivers
v0x12e3460_0 .net "curr_value", 31 0, v0x12e2de0_0;  alias, 1 drivers
v0x12e3530_0 .net "gated_new_value", 31 0, L_0x12f3de0;  1 drivers
v0x12e3620_0 .net "new_value", 31 0, o0x7f77672784c8;  alias, 0 drivers
L_0x12f3de0 .functor MUXZ 32, o0x7f77672784c8, L_0x7f776722f018, o0x7f77672781f8, C4<>;
S_0x12e2a60 .scope module, "r" "register" 3 19, 4 11 0, S_0x12e2800;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "should_write"
    .port_info 2 /INPUT 32 "new_value"
    .port_info 3 /OUTPUT 32 "curr_value"
v0x12e2cd0_0 .net "clock", 0 0, o0x7f7767278018;  alias, 0 drivers
v0x12e2de0_0 .var "curr_value", 31 0;
v0x12e2ec0_0 .var "is_init", 0 0;
v0x12e2f60_0 .net "new_value", 31 0, L_0x12f3de0;  alias, 1 drivers
L_0x7f776722f060 .functor BUFT 1, C4<1>, C4<0>, C4<0>, C4<0>;
v0x12e3040_0 .net "should_write", 0 0, L_0x7f776722f060;  1 drivers
    .scope S_0x12e2a60;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x12e2ec0_0, 0, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x12e2de0_0, 0, 32;
    %end;
    .thread T_0;
    .scope S_0x12e2a60;
T_1 ;
    %wait E_0x12ca520;
    %load/vec4 v0x12e2ec0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.0, 8;
    %load/vec4 v0x12e3040_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.2, 8;
    %load/vec4 v0x12e2f60_0;
    %assign/vec4 v0x12e2de0_0, 0;
T_1.2 ;
    %jmp T_1.1;
T_1.0 ;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0x12e2ec0_0, 0;
T_1.1 ;
    %jmp T_1;
    .thread T_1;
    .scope S_0x12bc9d0;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x12e1f20_0, 0, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x12e1e40_0, 0, 32;
    %end;
    .thread T_2;
    .scope S_0x12bc9d0;
T_3 ;
    %wait E_0x12ca520;
    %load/vec4 v0x12e1f20_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_3.0, 8;
    %load/vec4 v0x12e20d0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_3.2, 8;
    %load/vec4 v0x12e1ff0_0;
    %assign/vec4 v0x12e1e40_0, 0;
T_3.2 ;
    %jmp T_3.1;
T_3.0 ;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0x12e1f20_0, 0;
T_3.1 ;
    %jmp T_3;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "fetch_pipeline_reg.v";
    "./pipeline_reg.v";
    "./register.v";
