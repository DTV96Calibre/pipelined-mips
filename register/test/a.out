#! /usr/remote/bin/vvp
:ivl_version "10.1 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_0x1cd76b0 .scope module, "register_test" "register_test" 2 5;
 .timescale 0 0;
v0x1ceea70_0 .var "clock", 0 0;
v0x1ceeb30_0 .var "input_val", 31 0;
v0x1ceec00_0 .net "output_val", 31 0, v0x1cee650_0;  1 drivers
v0x1ceed00_0 .var "should_write", 0 0;
S_0x1cd7830 .scope module, "test" "register" 2 39, 3 11 0, S_0x1cd76b0;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clock"
    .port_info 1 /INPUT 1 "should_write"
    .port_info 2 /INPUT 32 "new_value"
    .port_info 3 /OUTPUT 32 "curr_value"
v0x1cc7980_0 .net "clock", 0 0, v0x1ceea70_0;  1 drivers
v0x1cee650_0 .var "curr_value", 31 0;
v0x1cee730_0 .var "is_init", 0 0;
v0x1cee800_0 .net "new_value", 31 0, v0x1ceeb30_0;  1 drivers
v0x1cee8e0_0 .net "should_write", 0 0, v0x1ceed00_0;  1 drivers
E_0x1cda3b0 .event posedge, v0x1cc7980_0;
    .scope S_0x1cd7830;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1cee730_0, 0, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x1cee650_0, 0, 32;
    %end;
    .thread T_0;
    .scope S_0x1cd7830;
T_1 ;
    %wait E_0x1cda3b0;
    %load/vec4 v0x1cee730_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.0, 8;
    %load/vec4 v0x1cee8e0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.2, 8;
    %load/vec4 v0x1cee800_0;
    %assign/vec4 v0x1cee650_0, 0;
T_1.2 ;
    %jmp T_1.1;
T_1.0 ;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0x1cee730_0, 0;
T_1.1 ;
    %jmp T_1;
    .thread T_1;
    .scope S_0x1cd76b0;
T_2 ;
    %vpi_call 2 14 "$dumpfile", "register_test.dump" {0 0 0};
    %vpi_call 2 15 "$dumpvars", 32'sb00000000000000000000000000000000, v0x1ceea70_0, v0x1ceeb30_0, v0x1ceec00_0, v0x1ceed00_0 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %pushi/vec4 100, 0, 32;
    %store/vec4 v0x1ceeb30_0, 0, 32;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceed00_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %pushi/vec4 5, 0, 32;
    %store/vec4 v0x1ceeb30_0, 0, 32;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceed00_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %pushi/vec4 44, 0, 32;
    %store/vec4 v0x1ceeb30_0, 0, 32;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x1ceea70_0, 0, 1;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "register_test.v";
    "./../register.v";
